app/src/main/java/io/github/droidkaigi/confsched/activity/BaseActivity.java

```diff
@@ -0,0 +1,22 @@
+package io.github.droidkaigi.confsched.activity;
+
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+
+import io.github.droidkaigi.confsched.MainApplication;
+import io.github.droidkaigi.confsched.di.ActivityComponent;
+import io.github.droidkaigi.confsched.di.ActivityModule;
+
+public abstract class BaseActivity extends AppCompatActivity {
+
+    private ActivityComponent activityComponent;
+
+    @NonNull
+    public ActivityComponent getComponent() {
+        if (activityComponent == null) {
```

Comment:
Attached fragments will call this method on their `getComponent()` (and so it is public).
`activityComponent` should be a *singleton in the activity scope*. So I created the base classes and added this methods.